pkg/ingestor/verifier/sigstore_verifier/sigstore_verifier_test.go

```diff
@@ -0,0 +1,393 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sigstore_verifier
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"math/rand"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/ingestor/key"
+	"github.com/guacsec/guac/pkg/ingestor/verifier"
+	"github.com/in-toto/in-toto-golang/in_toto"
+	slsa "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.2"
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/sigstore/sigstore/pkg/signature/dsse"
+)
+
+type mockKeyProvider struct {
+	collector map[string]key.Key
+}
+
+func newMockProvider() *mockKeyProvider {
+	return &mockKeyProvider{
+		collector: map[string]key.Key{},
+	}
+}
+
+func (m *mockKeyProvider) RetrieveKey(id string) (*key.Key, error) {
+	if key, ok := m.collector[id]; ok {
+		return &key, nil
+	}
+	return nil, nil
```

Comment:
hmm , on 2nd thought, maybe the interface should be err with a custom NotFound error to check against... but it's fine for now... just crossed my mind, not an issue since its documented in the interface.

Reply:
so the thought process here was that if it is not found by the `RetrieveKey` it would move on to the next key provider when using the `Find`. When `Retrieve` is called, it would return an error: `errors.New("failed to find key from key provider")`